query-builder/src/main/java/com/datastax/oss/driver/internal/querybuilder/insert/DefaultInsert.java

```diff
@@ -192,6 +220,14 @@ public String asCql() {
         builder.append(timestamp);
       }
     }
+    if (ttl != null) {
+      builder.append(" USING TTL ");
```

Comment:
If the timestamp was also specified, this must be `"  AND TTL "`.

Since there are only two possible clauses I don't think we need to get too fancy, we can just re-check `timestamp != null` to determine which keyword to use.